libs/application-generic/src/services/resource-validator.service.ts

```diff
@@ -0,0 +1,85 @@
+import { Injectable } from '@nestjs/common';
+import {
+  EnvironmentRepository,
+  NotificationTemplateRepository,
+} from '@novu/dal';
+import { FeatureFlagsKeysEnum } from '@novu/shared';
+
+import { GetFeatureFlag, NotificationStep } from '../usecases';
+import { ApiException } from '../utils/exceptions';
+
+@Injectable()
+export class ResourceValidatorService {
+  private readonly MAX_STEPS_PER_WORKFLOW = 10;
+  private readonly MAX_WORKFLOWS_LIMIT = 1000;
+
+  constructor(
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private environmentRepository: EnvironmentRepository,
+    private getFeatureFlag: GetFeatureFlag,
+  ) {}
+
+  async validateStepsLimit(environmentId: string, steps: NotificationStep[]) {
+    const isWorkflowLimitEnabled = await this.getFeatureFlag.execute({
+      key: FeatureFlagsKeysEnum.IS_MAX_WORKFLOW_LIMIT_ENABLED,
+      environmentId,
+      organizationId: 'system',
+      userId: 'system',
+    });
+
+    if (!isWorkflowLimitEnabled) {
+      return;
+    }
+
+    if (steps.length > this.MAX_STEPS_PER_WORKFLOW) {
+      throw new ApiException({
+        message: `Workflow steps limit exceeded. Maximum allowed steps is ${this.MAX_STEPS_PER_WORKFLOW}, but got ${steps.length} steps.`,
+        providedStepsCount: steps.length,
+        maxSteps: this.MAX_STEPS_PER_WORKFLOW,
+      });
+    }
+  }
+
+  async validateWorkflowLimit(environmentId: string) {
+    const environment = await this.getEnvironment(environmentId);
+
+    const workflowsCount = await this.notificationTemplateRepository.count({
+      _environmentId: environmentId,
+    });
```

Comment:
I think that this should be an environment-specific setting as it will allow us to define different numbers to different clients, same about the steps.
cc @SokratisVidros

Reply:
if it is environment-specific, we allow to have X amount of workflows per environment, did I miss something?

for steps, the limit is X amount of steps per workflow, in the same environment.

Reply:
We don't have the content for subscriptionId yet in Mongo. So for now the content in LD is user, org, env.

Reply:
ok... we should also align it with what @tatarco is doing for the new tier here: https://github.com/novuhq/novu/pull/7705/files in file `apps/api/src/app/environments-v1/usecases/create-environment/validate-tiers-use.case.ts`